diff --git a/plat/allwinner/common/include/platform_def.h b/plat/allwinner/common/include/platform_def.h
index c9d075ac8..20c398183 100644
--- a/plat/allwinner/common/include/platform_def.h
+++ b/plat/allwinner/common/include/platform_def.h
@@ -39,9 +39,9 @@
 
 #define SUNXI_BL33_VIRT_BASE		SUNXI_DRAM_VIRT_BASE
 
-/* The SCP firmware is allocated the last 16KiB of SRAM A2. */
+/* The SCP firmware is allocated the last 28KiB of SRAM A2. */
 #define SUNXI_SCP_BASE			BL31_LIMIT
-#define SUNXI_SCP_SIZE			0x4000
+#define SUNXI_SCP_SIZE			0x7000
 
 #endif /* SUNXI_BL31_IN_DRAM */
 
diff --git a/plat/allwinner/common/include/sunxi_def.h b/plat/allwinner/common/include/sunxi_def.h
index 20f6c4986..16ff7259f 100644
--- a/plat/allwinner/common/include/sunxi_def.h
+++ b/plat/allwinner/common/include/sunxi_def.h
@@ -9,6 +9,7 @@
 
 /* Clock configuration */
 #define SUNXI_OSC24M_CLK_IN_HZ		24000000
+#define SUNXI_PERIPH0_CLK_IN_HZ 	300000000
 
 /* UART configuration */
 #define SUNXI_UART0_BAUDRATE		115200
diff --git a/plat/allwinner/common/sunxi_bl31_setup.c b/plat/allwinner/common/sunxi_bl31_setup.c
index a32124a18..cd93e0f0b 100644
--- a/plat/allwinner/common/sunxi_bl31_setup.c
+++ b/plat/allwinner/common/sunxi_bl31_setup.c
@@ -81,6 +81,10 @@ static void sunxi_find_dtb(void)
 void bl31_early_platform_setup2(u_register_t arg0, u_register_t arg1,
 				u_register_t arg2, u_register_t arg3)
 {
+
+	/* Setup CPUS frequency to 300 MHz  / PERIPH0*/
+	mmio_write_32(SUNXI_R_PRCM_BASE, 2<<16 | 1<<8);
+
 	/* Initialize the debug console as soon as possible */
 	console_16550_register(SUNXI_UART0_BASE, SUNXI_UART0_CLK_IN_HZ,
 			       SUNXI_UART0_BAUDRATE, &console);
diff --git a/plat/allwinner/sun50i_a64/sunxi_power.c b/plat/allwinner/sun50i_a64/sunxi_power.c
index a35b9ddc0..1645eab39 100644
--- a/plat/allwinner/sun50i_a64/sunxi_power.c
+++ b/plat/allwinner/sun50i_a64/sunxi_power.c
@@ -93,8 +93,8 @@ static int rsb_init(void)
 	if (ret)
 		return ret;
 
-	/* Switch to the recommended 3 MHz bus clock. */
-	ret = rsb_set_bus_speed(SUNXI_OSC24M_CLK_IN_HZ, 3000000);
+	/* Switch to 300 MHz bus clock. */
+	ret = rsb_set_bus_speed(SUNXI_PERIPH0_CLK_IN_HZ, 3000000);
 	if (ret)
 		return ret;
 
